nav2_dynamic_params/src/example_dynamic_params.cpp

```diff
@@ -64,22 +65,53 @@ int main(int argc, char ** argv)
   param_validator->add_param("foo", rclcpp::ParameterType::PARAMETER_DOUBLE);
   param_validator->add_param("bar", rclcpp::ParameterType::PARAMETER_INTEGER, {0, 10});
 
+  // Create a custom validation callback
+  std::function<rcl_interfaces::msg::SetParametersResult(
+      const std::vector<rclcpp::Parameter> &)> custom_validation_callback = [node](
+    const std::vector<rclcpp::Parameter> & parameters)
+    -> rcl_interfaces::msg::SetParametersResult
+    {
```

Comment:
Long functions like this a better as named functions, not lambdas.

Reply:
Right. I made them functions outside of main